document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/pscs/ApiToPscMapper.java

```diff
@@ -0,0 +1,125 @@
+            @Mapping(target = "dateOfBirth", ignore = true),
+            @Mapping(target = "ceasedOn", ignore = true),
+            @Mapping(target = "notifiedOn", ignore = true)
+    })
+
+    public abstract Psc apiToPsc(PscApi pscApi) throws MapperException;
+
+    public abstract List<Psc> apiToPsc(List<PscApi> pscApi) throws MapperException;
+
+    @Autowired
+    private RetrieveApiEnumerationDescription retrieveApiEnumerationDescription;
+
+    @AfterMapping
+    protected void setNaturesOfControl(PscApi pscApi, @MappingTarget Psc psc) {
+
+        if (pscApi != null && pscApi.getNaturesOfControl() != null) {
+            psc.setNaturesOfControl(setNaturesOfControl(pscApi.getNaturesOfControl()));
+        }
+    }
+
+    @AfterMapping
+    protected void setCeasedOnDate(PscApi pscApi, @MappingTarget Psc psc) {
+
+        if (pscApi != null && pscApi.getCeasedOn() != null) {
+            LocalDate ceasedOn = pscApi.getCeasedOn();
+            psc.setCeasedOn(ceasedOn.format(getFormatter()));
+        }
+    }
+
+    @AfterMapping
+    protected void setDateOfBirth(PscApi pscApi, @MappingTarget Psc psc) {
+
+        if (pscApi != null && pscApi.getDateOfBirth() != null) {
+            DateOfBirth dob = new DateOfBirth();
+            String monthString = getNameOfMonth(pscApi);
+
+            dob.setYear(pscApi.getDateOfBirth().getYear());
+            //Sentence case month string
+            dob.setMonth(monthString.substring(0, 1).toUpperCase()
+                    + monthString.substring(1).toLowerCase());
+
+            psc.setDateOfBirth(dob);
+        }
+    }
+
+    @AfterMapping
+    protected void setNotifiedOn(PscApi pscApi, @MappingTarget Psc psc) {
+
+        if (pscApi != null && pscApi.getNotifiedOn() != null) {
+            LocalDate notifiedOn = pscApi.getNotifiedOn();
+            psc.setNotifiedOn(notifiedOn.format(getFormatter()));
+        }
+    }
+
+    private List<NaturesOfControl> setNaturesOfControl(String[] naturesOfControl) {
+
+        List<NaturesOfControl> naturesOfControlList = new ArrayList<>();
+
+        if (naturesOfControl != null) {
+            for (String natureOfControl : naturesOfControl) {
+                NaturesOfControl natures = new NaturesOfControl();
+                natures.setNaturesOfControl(natureOfControl);
+                String natureOfControlDescription = retrieveApiEnumerationDescription
+                        .getApiEnumerationDescription(PSC_DESCRIPTIONS, IDENTIFIER,
+                                natureOfControl, getDebugMap(natureOfControl));
+                natures.setNaturesOfControlDescription(natureOfControlDescription);
+                naturesOfControlList.add(natures);
+            }
+        }
+
+        return naturesOfControlList;
+    }
+
+    private Map<String, String> getDebugMap(String debugString) {
+
+        Map<String, String> debugMap = new HashMap<>();
+        debugMap.put(ENUMERATION_MAPPING, debugString);
```

Comment:
what does enumeration mapping show to link it to the request that this is for?

i.e how can this data help with debugging, can we link it to the company document that is being generated?

Reply:
Debugging and logging across the template will be looked at in PCI-134